doctest: pkgconfig adds `-ldoctest` when no such lib exists

### Nixpkgs version

- Unstable (26.05)

### Describe the bug

The doctest package builds a pkgconfig with the following line:

```
Libs: -L${libdir} -ldoctest
```


doctest is a single header library and does not contain any shared/static libs, thus this entry is wrong

### Steps to reproduce

on the master branch (4821782ec678ef791d52c05bba03dcdf0c83668a),

```
nix build .#doctest
```

### Expected behaviour

pkgconfig does not include `-ldoctest` flag

### Screenshots

_No response_

### Relevant log output

```console

```

### Additional context

here's the log output from building a package that depends on doctest:

```
error: Cannot build '/nix/store/rrda5n91lljx4v8x38d8ll3zhbh4ralg-wf-touch-git.drv'
.
       Reason: builder failed with exit code 1.
       Output paths:
         /nix/store/i22bqvh0ilf8gcqvq32gw6n86hyrnhwp-wf-touch-git-dev
         /nix/store/v3agyf8dixf43qiv8l1mfhi81nj4b3jf-wf-touch-git
       Last 25 log lines:
       >     libdir       : /nix/store/v3agyf8dixf43qiv8l1mfhi81nj4b3jf-wf-touch-g
it/lib
       >     libexecdir   : /nix/store/v3agyf8dixf43qiv8l1mfhi81nj4b3jf-wf-touch-g
it/libexec
       >     localedir    : /nix/store/v3agyf8dixf43qiv8l1mfhi81nj4b3jf-wf-touch-g
it/share/locale
       >     mandir       : /nix/store/v3agyf8dixf43qiv8l1mfhi81nj4b3jf-wf-touch-g
it/share/man
       >     prefix       : /nix/store/v3agyf8dixf43qiv8l1mfhi81nj4b3jf-wf-touch-g
it
       >     sbindir      : /nix/store/v3agyf8dixf43qiv8l1mfhi81nj4b3jf-wf-touch-g
it/sbin
       >     wrap_mode    : nodownload
       >
       > Found ninja-1.13.2 at /nix/store/vvg707j9xgklk7j4yqa8s8zxzpgwypsq-ninja-1
.13.2/bin/ninja
       > mesonConfigurePhase: enabled\ parallel\ building
       > Running phase: buildPhase
       > build flags: -j32
       > [1/10] Compiling C++ object libwftouch.a.p/src_touch.cpp.o
       > [2/10] Compiling C++ object libwftouch.a.p/src_actions.cpp.o
       > [3/10] Compiling C++ object libwftouch.a.p/src_math.cpp.o
       > [4/10] Linking static target libwftouch.a
       > [5/10] Compiling C++ object test/gesture_test.p/gesture_test.cpp.o
       > [6/10] Linking target test/gesture_test
       > FAILED: [code=1] test/gesture_test
       > g++  -o test/gesture_test test/gesture_test.p/gesture_test.cpp.o -Wl,--as
-needed -Wl,--no-undefined -Wl,-O1 -Wl,--start-group libwftouch.a -L/nix/store/k3x
mbic33n5gbd68ss6ll4rsc74mv9jx-doctest-2.5.0/lib -ldoctest -Wl,--end-group
       > /nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: can
not find -ldoctest: No such file or directory
       > collect2: error: ld returned 1 exit status
       > [7/10] Compiling C++ object test/action_test.p/action_test.cpp.o
       > [8/10] Compiling C++ object test/basic_test.p/basic_test.cpp.o
       > ninja: build stopped: subcommand failed.
       For full logs, run:
         nix log /nix/store/rrda5n91lljx4v8x38d8ll3zhbh4ralg-wf-touch-git.drv
error: Cannot build '/nix/store/6l5c300397cmfqs4zy62h02h1macypwv-hyprgrass-0.8.2+d
irty.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/vs4h7xy8ymn9lxm1rk47qda0mz05zkda-hyprgrass-0.8.2+dirty
```

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.80, NixOS, 25.11 (Xantusia), 25.11.20260410.54170c5`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.4`
 - nixpkgs: `/nix/store/m5dr0w2wb9lca77ar3idab31szd6mix7-source`


### Notify maintainers


maintainer ping @nim65s 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/do/doctest/package.nix
+++ b/pkgs/by-name/do/doctest/package.nix
@@ -7,13 +7,13 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "doctest";
-  version = "2.5.0";
+  version = "2.5.2";
 
   src = fetchFromGitHub {
     owner = "doctest";
     repo = "doctest";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-7t/eknv7VtHoBgcuJmI07x//HIyqzE9HUuH5u2y7X8A=";
+    hash = "sha256-4jW6xPFCFxk1l47EkSUVojhycrtluPhOc5Adf/25R7M=";
   };
 
   nativeBuildInputs = [ cmake ];
@@ -27,6 +27,7 @@ stdenv.mkDerivation (finalAttrs: {
   doCheck = true;
 
   meta = {
+    changelog = "https://github.com/doctest/doctest/releases/tag/${finalAttrs.src.tag}";
     homepage = "https://github.com/doctest/doctest";
     description = "Fastest feature-rich C++11/14/17/20 single-header testing framework";
     platforms = lib.platforms.all;

